speechbrain/lobes/models/huggingface_whisper.py

```diff
@@ -155,10 +173,16 @@ def forward_encoder(self, wav):
 
         if self.freeze_encoder:
             with torch.no_grad():
-                mel = self._get_mel(wav)
-                return self.model.encoder(mel).last_hidden_state
+                return self._get_encoder_states(wav)
+        else:
+            return self._get_encoder_states(wav)
+
+    def _get_encoder_states(self, wav):
+        mel = self._get_mel(wav)
+        if self.output_all_hiddens:
+            states = self.model.encoder(mel, output_hidden_states=True)
+            return torch.stack(states.hidden_states)
```

Comment:
is it the same to `torch.stack(list(states.hidden_states), dim=0)` (taken from wav2vec2)?

Reply:
Yes! I think it is not necessary to take this extra step to first convert to a list, we can stack the tuples directly.